pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/InstanceDecommissionIntegrationTest.java

```diff
@@ -0,0 +1,173 @@
+    try {
+      sendDeleteRequest(deleteInstanceRequest);
+      Assert.fail("Delete should have returned a failure status (409)");
+    } catch (IOException e) {
+      // Expected exception because of 409 status code
+    }
+
+    // Delete the table
+    sendDeleteRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forDeleteTableWithType("mytable", "offline")
+    );
+
+    // Now, decommissioning for the server should work
+    response = new JSONObject(sendDeleteRequest(deleteInstanceRequest));
+    System.out.println(response);
+    Assert.assertEquals(response.get("status").toString(), "success");
+
+    // Try to delete a broker whose information is still live
+    try {
+      deleteInstanceRequest = ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forInstanceDelete(brokerName);
+      sendDeleteRequest(deleteInstanceRequest);
+      Assert.fail("Delete should have returned a failure status (409)");
+    } catch (IOException e) {
+      // Expected exception because of 409 status code
+    }
+
+    // TODO: Add test to delete broker instance. Currently, stopBroker() does not work correctly.
+
+    // Check if '/INSTANCES/<server_name>' has been erased correctly
+    ensureZkHelixManagerIsInitialized();
+    String instancePath = "/" + _zkHelixManager.getClusterName() + "/INSTANCES/" + serverName;
+    boolean exist = _zkHelixManager.getHelixDataAccessor().getBaseDataAccessor()
+        .exists(instancePath, AccessOption.PERSISTENT);
+    Assert.assertEquals(exist, false);
+
+    // Check if '/CONFIGS/PARTICIPANT/<server_name>' has been erased correctly
+    String configPath = "/" + _zkHelixManager.getClusterName() + "/CONFIGS/PARTICIPANT/" + serverName;
+    exist = _zkHelixManager.getHelixDataAccessor().getBaseDataAccessor().exists(instancePath, AccessOption.PERSISTENT);
+    Assert.assertEquals(exist, false);
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testDistinctCountGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testDistinctCountNoGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testDistinctCountHLLNoGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testDistinctCountHLLGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testQuantileGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testQuantileNoGroupByQuery() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testHardcodedQuerySet() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testGeneratedQueriesWithMultiValues() throws Exception {
+  }
+
+  @Override
+  @Test(enabled = false)
+  public void testInstancesStarted() {
+  }
+
+  @AfterClass
+  @Override
+  public void tearDown() throws Exception {
+    try {
```

Comment:
I'd just call the appropriate methods (stopBroker, etc.) instead of trapping the exception.